docs/doxygen/pages.dox

```diff
@@ -0,0 +1,99 @@
+
+ The AWS IoT Device Defender Client Library provides a convenience API for interacting with AWS IoT Device Defender service. This library is independent of the MQTT library. Applications can use this library to assemble and parse the Device Defender MQTT topics, then use any MQTT library to publish/subscribe to those topics. Features of this library include:
+- It is stateless.  It does not use any global/static memory.
+- It depends only on the standard C library only.
+
+@section defender_memory_requirements Memory Requirements
+@brief Memory requirements of the AWS IoT Device Defender Client Library.
+
+<table>
+    <tr>
+        <td colspan="4"><center><b>Code size of AWS IoT Device Defender Client Library files (sizes generated with [GCC for ARM Cortex-M toolchain 20191025](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads/9-2019-q4-major))</b></center></td>
+    </tr>
+    <tr>
+        <td><b>File</b></td>
+        <td><b>No Optimization (asserts enabled)</b></td>
+        <td><b>With -O1 Optimization (asserts disabled)</b></td>
+        <td><b>With -Os Optimization (asserts disabled)</b></td>
+    </tr>
+    <tr>
+        <td>defender.c</td>
+        <td>2.0K</td>
+        <td>1.1K</td>
+        <td>0.6K</td>
+    </tr>
+</table>
+ */
+
+/**
+@page defender_design Design
+AWS IoT Device Defender Client Library Design
+
+The AWS IoT Device Defender Client Library provides macros and functions to assemble and parse MQTT topic strings reserved for the Device Defender. It does not depend on MQTT library and the application can use any MQTT library to interact with the AWS IoT Device Defender service.
+
+The diagram below demonstrates how an application uses the Device Defender Client Library, an MQTT library, and a JSON library to interact with the AWS IoT Device Defender service.
+
+\image html defender_design_operations.png "Device Defender Client Library Demo Operation Diagram" width=1000px
+*/
+
+/**
+@page defender_config Configurations
+@brief Configurations of the AWS IoT Device Defender Client Library.
+<!-- @par configpagestyle allows the @section titles to be styled according to style.css -->
+@par configpagestyle
+
+Configuration settings are C pre-processor constants. They can be set with a `\#define` in the config file (`defender_config.h`) or by using a compiler option such as -D in gcc.
+
+@section DEFENDER_DO_NOT_USE_CUSTOM_CONFIG
+@copydoc DEFENDER_DO_NOT_USE_CUSTOM_CONFIG
+
+@section DEFENDER_USE_LONG_KEYS
+@copydoc DEFENDER_USE_LONG_KEYS
+
+@section LogError
+@copydoc LogError
+
+@section LogWarn
+@copydoc LogWarn
+
+@section LogInfo
+@copydoc LogInfo
+
+@section LogDebug
+@copydoc LogDebug
```

Comment:
Can you add these to avoid section name clashes in the CSDK? This is a detail I need to put in the template instructions.

```suggestion
@section defender_logerror LogError
@copydoc LogError

@section defender_logwarn LogWarn
@copydoc LogWarn

@section defender_loginfo LogInfo
@copydoc LogInfo

@section defender_logdebug LogDebug
@copydoc LogDebug
```

Reply:
Sure, added.